packages/ozone/src/mod-service/index.ts

```diff
@@ -1266,6 +1268,130 @@ export class ModerationService {
       throw new InvalidRequestError('Email was accepted but not sent')
     }
   }
+
+  buildModerationQuery(
```

Comment:
I am not too found of writing this as a query builder. I find the query hard to read. More importantly, this causes the types not to match reality, causing un-necessary runtime overhead.

In the following loop, the `??` is required because the the type is too lax but the value will never be undefined at runtime, causing an un-necessary nullish check.
```ts
    // Merge accountReports
    for (const report of accountReports) {
      const entry = ensureDidEntry(report.did)
      entry.accountReportCount = report.accountReportCount ?? 0
      entry.reportedAccountCount = report.reportedAccountCount ?? 0
    }
```